python/doc/examples/graphs/plot_graphs_fill_area.py

```diff
@@ -49,34 +49,66 @@ def linearSample(xmin, xmax, npoints):
 # The following function uses the `PolygonArray` class to create a area filled with a given color.
 
 # %%
-def drawInTheBounds(vLow, vUp, n_test):
+def fillWithColorInBetween(x, yLower, yUpper, color):
     """
-    Draw the area within the bounds.
+    Fill an area with a color between two bounds
+
+    Parameters
+    ----------
+    x : ot.Point(n_points)
+        The X coordinates.
+    yLower : ot.Point(n_points)
+        The Y coordinate of the lower bound.
+    yUpper : ot.Point(n_points)
+        The Y coordinate of the upper bound.
+
+    Return
+    ------
+    polygonArray : ot.PolygonArray()
+        The colored area.
     """
-    palette = ot.Drawable.BuildDefaultPalette(2)
-    myPaletteColor = palette[1]
-    polyData = [[vLow[i], vLow[i + 1], vUp[i + 1], vUp[i]] for i in range(n_test - 1)]
+    n_points = x.getDimension()
+    if yLower.getDimension() != n_points:
+        raise ValueError(
+            f"The number of Y lower bounds is equal to {yLower.getDimension()} "
+            f"but the number of X coordinates is {n_points}"
+        )
+    if yUpper.getDimension() != n_points:
+        raise ValueError(
+            f"The number of Y upper bounds is equal to {yUpper.getDimension()} "
+            f"but the number of X coordinates is {n_points}"
+        )
+    n_points = x.getDimension()
+    polyData = []
```

Comment:
Maybe
```
polydata = [[x[i], yLower[i]] for i in range(n_points)] + \
            [[x[i], yUpper[i]] for i in range(n_points-1, -1, -1)]
```
 to avoid loops.